packages/flutter_solidart/lib/src/widgets/solid.dart

```diff
@@ -246,21 +246,25 @@ part '../models/solid_element.dart';
 /// ```
 /// {@endtemplate}
 @immutable
-class Solid extends StatefulWidget {
-  /// {@macro solid}
-  const Solid({
+class ProviderScope extends StatefulWidget {
+  /// {@macro provider-scope}
+  const ProviderScope({
     super.key,
     this.child,
+    this.providers = const [],
+  })  : builder = null,
+        _canAutoDisposeProviders = true;
+
+  /// {@macro provider-scope}
+  const ProviderScope.builder({
+    super.key,
     this.builder,
```

Comment:
I don't like the idea to separate the builder from the main constructor.
At the end, they are just two different methods to pass a child.
If you use the builder function you get immediately a context which can see the provided providers, with child not.